db/db.go

```diff
@@ -160,6 +174,13 @@ func (b *boltDB) PutIfNotExists(namespace string, key []byte, value []byte) erro
 		}
 		return ErrAlreadyExist
 	})
+
+	if err != nil && c < quota {
```

Comment:
If the error is ErrAlreadyExist, i think we don't want to retry, because it's a business logic error. We only should retry on db transit issue.